express-notes-api/index.js

```diff
@@ -0,0 +1,198 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const inputFileName = 'data.json'
+
+// Clients can GET a list of notes.
+
+app.get('/api/notes', (req, res) => {
+
+    fs.readFile(inputFileName, 'utf8', (err, data) => {
+        if (err) {
+            console.log("Error reading File:",  err);
+        } try {
+            const content = JSON.parse(data);
+            const outArray = [];
+            for (const outObj in content.notes) {
+                outArray.push({
+                    "content": content.notes[outObj].content,
+                    "id": content.notes[outObj].id
+                })
+            }
+            res.status(200)
+            res.json(outArray);
+        } catch (err) {
+            console.log("Error parsing JSON string: ", err);
+        } 
+    });
+})
+
+
+// Clients can GET a single note by id.
+
+
+app.get('/api/notes/:id', (req, res) => {
+    if (parseInt(req.params.id) <= 0 &&  typeof req.params.id !== 'number' && req.params.id % 1 !== 0) {
+        res.status(400);
+        res.json({"error": "id must be a positive number"})
+    } else {
+        fs.readFile(inputFileName, 'utf8', (err, data) => {
+            if (err) {
+                console.log("Error reading File:",  err);
+            } try {
+                const content = JSON.parse(data);
+                const outArray = [];
+                for (const outObj in content.notes) {
+                    if (content.notes[outObj].id === parseInt(req.params.id)) {
+                        outArray.push({
+                            "content": content.notes[outObj].content,
+                            "id": content.notes[outObj].id
+                        })
+                    }
+                }
+                if (outArray.length > 0) {
+                    res.status(200);
+                    res.json(outArray);
+                } else {

```

Comment:
The success response for getting a single note should not be wrapped in an array:

---

![image](https://user-images.githubusercontent.com/16143771/130301042-3ddf02c6-96ba-4e1c-9bc7-6ede38a332c3.png)

Reply:
Change the response for single id request to send only the JSON for the id without an array object.